flake.nix

```diff
@@ -0,0 +1,63 @@
+{
+  inputs = {
+    flake-utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+  };
+
+  outputs = { self, flake-utils, rust-overlay, nixpkgs }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        overlays = [ (import rust-overlay) ];
+        pkgs = (import nixpkgs) { inherit system overlays; };
+      in
+      {
+        # Provide a shell with all dependencies needed to build `koragnar::*`
+        devShell = pkgs.mkShell
+          {
+            nativeBuildInputs = with pkgs; [
+              (rust-bin.selectLatestNightlyWith (toolchain: toolchain.default.override {
```

Comment:
I was trying to find some refenrence on `selectLatestNightlyWith` to figure out if it respects the `rust-toolchain.toml`, but I believe it does not, correct?